### DISCUSSION TITLE
Argocd Comparison Error while deploying with helm

### DISCUSSION THREAD

**Discussion by prabormukherjee** [argoproj/argo-cd#7296] [Q&A]
I am trying to deploy an argocd application, but got the following error
```ComparisonErrorrpc error: code = Unknown desc = Get "https://github.com/prabormukherjee/Techtrends/info/refs?service=git-upload-pack": dial tcp: lookup github.com on 10.43.0.10:53: server misbehaving```   

The build was successful: 
`$ kubectl apply -f argocd/helm-techtrends-prod.yaml`
>application.argoproj.io/techtrends-prod created  

`kubectl get application -n argocd`
>NAME   SYNC STATUS   HEALTH STATUS
techtrends-prod   Unknown   Healthy  

I run these commands for debugging from multiple suggestions given in knowledge. Here is the [file](https://ufile.io/sc03w92u), you can download.

If you want to check any yaml file, you can find them [here](https://github.com/prabormukherjee/Techtrends). For `yaml` of argocd, check the `argocd/`. There I have `helm-techtrends-prod.yaml` and `helm-techtrends-staging.yaml`, that I want to deploy.

**Comment by ishitasequeira** [comment]
@prabormukherjee, It seems the error is due to dns hostname resolution failure. Try checking the dns hostname resolution for your network.

**Reply by prabormukherjee** [reply]
Thanks @ishitasequeira for your reply. Can you please tell me the steps to resolve? I don't have problems with internet, firewall or proxy.

**Reply by jgwest** [reply]
Agree with @ishitasequeira , it appears your in-cluster DNS resolver is not able to resolve `github.com` for unknown reasons (for instance, the port `53`referenced in the error message is the DNS port). This strongly suggests an issue with your K8s cluster setup, which we probably aren't equipped to solve here :smile:. 

The only thing I can think of that could _possibly_ cause this behave in Argo CD is if you were using the Argo CD repo proxy functionality (https://argoproj.github.io/argo-cd/operator-manual/declarative-setup/#configure-repositories-with-proxy), so if you are using that that might be something to check out (but you would know if you were, it's not enabled by default.)

**Reply by prabormukherjee** [reply]
Thanks @jgw. No I'm not using as proxy as such. That leaves with only the k3s setup. I'm using coredns which i think i the default for k3s. Not sure if using kubedns will help or not.
